app/modules/parsing/services/inference_cache_service.py

```
@@ -0,0 +1,148 @@
+            self.db.rollback()
+            logger.debug(
+                f"Race condition detected for content_hash: {content_hash[:12]}..., fetching existing entry"
+            )
+            cache_entry = (
+                self.db.query(InferenceCache)
+                .filter(InferenceCache.content_hash == content_hash)
+                .first()
+            )
+            if not cache_entry:
+                # This shouldn't happen, but re-raise if it does
+                raise e
```

Ruff (0.14.4)

Use `raise` without specifying exception name

Remove exception name

(TRY201)